src/Illuminate/Foundation/helpers.php

```diff
@@ -506,6 +506,24 @@ function factory()
     }
 }
 
+if (! function_exists('generic_asset')) {
+    /**
+     * Generate an asset path for the application acording to the APP_ENV value used.
+     *
+     * @param  string  $path
+     * @param  string  $env_value
+     * @return string
+     */
+    function generic_asset($path, $env_value = 'local')
+    {
+        if (env('APP_ENV', '') === $env_value) {
```

Comment:
You cant use `env()` here, because it'll break when the config is cached.

Reply:
I got it. Do you think calling getenv() instead of env() would cause the same problem?